src/Spanner/Database.php

```diff
@@ -0,0 +1,1313 @@
+     * possible. Do not hold transactions open longer than necessary.
+     *
+     * Example:
+     * ```
+     * $transaction = $database->transaction();
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.v1#google.spanner.v1.BeginTransactionRequest BeginTransactionRequest
+     * @see https://cloud.google.com/spanner/docs/transactions Transactions
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $options [optional] Configuration Options.
+     * @return Transaction
+     */
+    public function transaction(array $options = [])
+    {
+        // There isn't anything configurable here.
+        $options['transactionOptions'] = [
+            'readWrite' => []
+        ];
+
+        $session = $this->selectSession(SessionPoolInterface::CONTEXT_READWRITE);
+
+        try {
+            return $this->operation->transaction($session, $options);
+        } finally {
+            $session->setExpiration();
+        }
+    }
+
+    /**
+     * Insert a row.
+     *
+     * Mutations are committed in a single-use transaction.
+     *
+     * Example:
+     * ```
+     * $database->insert('Posts', [
+     *     'ID' => 1337,
+     *     'postTitle' => 'Hello World!',
+     *     'postContent' => 'Welcome to our site.'
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.v1#google.spanner.v1.CommitRequest CommitRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param string $table The table to mutate.
+     * @param array $data The row data to insert.
+     * @param array $options [optional] Configuration options.
+     * @return Timestamp The commit Timestamp.
+     */
+    public function insert($table, array $data, array $options = [])
+    {
+        return $this->insertBatch($table, [$data], $options);
+    }
+
+    /**
+     * Insert multiple rows.
+     *
+     * Mutations are committed in a single-use transaction.
+     *
+     * Example:
+     * ```
+     * $database->insertBatch('Posts', [
+     *     [
+     *         'ID' => 1337,
+     *         'postTitle' => 'Hello World!',
+     *         'postContent' => 'Welcome to our site.'
+     *     ], [
+     *         'ID' => 1338,
+     *         'postTitle' => 'Our History',
+     *         'postContent' => 'Lots of people ask about where we got started.'
+     *     ]
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.v1#google.spanner.v1.CommitRequest CommitRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param string $table The table to mutate.
+     * @param array $dataSet The row data to insert.
+     * @param array $options [optional] Configuration options.
+     * @return Timestamp The commit Timestamp.
+     */
+    public function insertBatch($table, array $dataSet, array $options = [])
```

Comment:
The goal would be to prevent the extra RPC call to create a transaction if they only need to insert to multiple tables, for example. Would it be preferred we attempt to infer whether we need single use or not, instead of allowing a user to specify?